backend/document/markdown_extensions/remove_section_preprocessor.py

```diff
@@ -27,28 +27,17 @@ def remove_sections(self, md: str) -> list[str]:
             md = self.remove_md_section(md, section)
         return md.split("\n")
 
-    def remove_md_section(self, md: str, section_name: str) -> str:
-        """
-        Given markdown and a section name, removes the section header and the
-        text contained in the section.
-        """
-        header_regex = re.compile("^#.*$")
-        section_regex = re.compile("^#+ {}".format(section_name))
-        out_md = ""
-        in_section = False
-        for line in md.splitlines():
-            if in_section:
-                if header_regex.match(line):
-                    # We found a header.  The section is over.
-                    out_md += line + "\n"
-                    in_section = False
-            else:
-                if section_regex.match(line):
-                    # We found the section header.
-                    in_section = True
-                else:
-                    out_md = "{}{}\n".format(out_md, line)
-        return out_md
+    def remove_md_section(
+        self,
+        md: str,
+        section_name: str,
+        # See https://regex101.com/r/SSItAG/1
+        section_regex_fmt_str: str = r"\#+ +{}.*$\n*.*",
```

Comment:
This regex seems like it would match the section header and the rest of the document, not just the rest of the section.  Does this have any tests that demonstrate correct behavior?

Reply:
Good catch, thank you!

Reply:
I updated this PR with quite a few changes. Here is the commit message for the latest push to this PR:

"Refactor remove_section_preprocessor to follow the pattern that the
other markdown preprocessor plugins follow so that the parameters the
plugin needs are passed in at construction time for the sake of
referential transparency.

Existing test was improved to ensure that markdown sections which are
not to be removed and occur after sections that are to be removed
remain intact after processing by this preprocessor.

Regex approach used in remove_md_section was abandoned in favor of
legacy algo since regex complexity required to yield desired result is
arguably more complex to grok for newcomers. Edit: The regex you commented on did indeed extend past the markdown sections to remove. 
"